spec.emu

```diff
@@ -6,77 +6,172 @@
+            1. <ins>Let _moduleSourceObject_ be _module_.[[ModuleSourceObject]].</ins>
+            1. <ins>Perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _moduleSourceObject_ &raquo;).</ins>
+            1. <ins>Return ~unused~.</ins>
+          1. Let _loadPromise_ be _module_.LoadRequestedModules().
+          1. Let _rejectedClosure_ be a new Abstract Closure with parameters (_reason_) that captures _promiseCapability_ and performs the following steps when called:
+            1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _reason_ &raquo;).
+            1. Return ~unused~.
+          1. Let _onRejected_ be CreateBuiltinFunction(_rejectedClosure_, 1, *""*, &laquo; &raquo;).
+          1. Let _linkAndEvaluateClosure_ be a new Abstract Closure with no parameters that captures _module_, _promiseCapability_, and _onRejected_ and performs the following steps when called:
+            1. Let _link_ be Completion(_module_.Link()).
+            1. If _link_ is an abrupt completion, then
+              1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _link_.[[Value]] &raquo;).
+              1. Return ~unused~.
+            1. Let _evaluatePromise_ be _module_.Evaluate().
+            1. Let _fulfilledClosure_ be a new Abstract Closure with no parameters that captures _module_ and _promiseCapability_ and performs the following steps when called:
+              1. Let _namespace_ be GetModuleNamespace(_module_).
+              1. Perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _namespace_ &raquo;).
+              1. Return ~unused~.
+            1. Let _onFulfilled_ be CreateBuiltinFunction(_fulfilledClosure_, *""*, 0, &laquo; &raquo;).
+            1. Perform PerformPromiseThen(_evaluatePromise_, _onFulfilled_, _onRejected_).
+          1. Let _linkAndEvaluate_ be CreateBuiltinFunction(_linkAndEvaluateClosure_, *""*, 0, &laquo; &raquo;).
+          1. Perform PerformPromiseThen(_loadPromise_, _linkAndEvaluate_, _onRejected_).
+          1. Return ~unused~.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
   </emu-clause>
 </emu-clause>
 
-<emu-clause id="sec-ecmascript-language-scripts-and-modules">
+<emu-clause id="sec-ecmascript-language-scripts-and-modules" number="16">
   <h1>ECMAScript Language: Scripts and Modules</h1>
-  
+
   <emu-clause id="sec-modules">
     <h1>Modules</h1>
 
     <emu-clause id="sec-module-semantics">
       <h1>Module Semantics</h1>
 
-      <emu-clause id="sec-static-semantics-modulerequests" oldids="sec-module-semantics-static-semantics-modulerequests,sec-imports-static-semantics-modulerequests,sec-exports-static-semantics-modulerequests" type="sdo">
-        <h1>Static Semantics: ModuleRequests ( ): a List of Strings</h1>
+      <emu-clause id="sec-static-semantics-modulerequests" type="sdo">
+        <h1>
+          Static Semantics: ModuleRequests (
+            <ins>optional _recursiveOnly_: a Boolean</ins>
```

Comment:
Rather than overloading module requests, perhaps we could create another type of static semantic like `ReflectionRequests`.

Reply:
A `ReflectionRequests` doesn't really work, because for
```js
import module a from "a";
import module b from "b";
import "a";
```

we only want to include `b` in the list of "modules where we should not recurse into", and not `a`. So we always have to do "all the dependencies" − "all the normal dependencies".

If you prefer, I can copy-paste this SDO to create a new `RecursiveModuleRequests` instead of adding the parameter, but keeping them as a single SDO helps keeping them in sync.